### PR TITLE
Ryan/optional creds

### PR DESCRIPTION
**Pull request by ryanisnan**
This PR makes ServiceRequestFactory optionally take credentials or tokens. This theoretically allows anonymous users to make use of the ServiceRequestFactory.

### CODE REVIEW

File: spylib/request.py [hunk 1]
```diff
@@ -117,7 +140,34 @@ def make_service_request(
         **kwargs
     ):
         """
-        Cross service communication request, with an optional retry mechanism.
+        Makes a cross-service request with configurable timeouts and retry counts.
+
```
Comment (alekhrycaiko) [inline spylib/request.py]: the docs here are :fire:

File: spylib/request.py [hunk 2]
```diff
@@ -90,7 +86,9 @@ def __init__(
                 decode(self.access_token, self.secret, algorithms=[self.algorithm])
             except ExpiredSignatureError:
                 self.fetch_new_tokens()
-        else:
+        elif self.refresh_token is not None:
+            self.fetch_new_tokens()
+        elif self.uuid is not None and self.api_key is not None:
             self.fetch_new_tokens()
```
Comment (alekhrycaiko) [inline spylib/request.py]: 💭 looks good!

File: spylib/request.py (hunk 1 above)
Comment (ryanisnan) [reply spylib/request.py]: Thanks! I didn't add all docstrings yet - will do with time.

**Comment by ryanisnan** [comment]
@alekhrycaiko do you think we should bump the version? Probably a good habit to form.... slightly more annoying though :P

File: spylib/request.py [hunk 3]
```diff
@@ -101,8 +99,33 @@ def _set_refresh_token(self, refresh_token):
...
+        Returns:
+            bool: Whether or not the instance is capable of authenticating.
+        """
+        if self.refresh_token is not None:
+            return True
+        
+        if (self.uuid is not None) and (self.api_key is not None):
+            return True
+        
+        return False
```
Comment (alekhrycaiko) [inline spylib/request.py]: Would we want to return true in the case that a user has an `access_token`, but, no `refresh_token`, `uuid`, or `api_key`?

Comment (alekhrycaiko) [reply spylib/request.py]: actually nm, that won't matter here I think 👍

Comment (ryanisnan) [reply spylib/request.py]: I think that would be good if the method was `is_authenticated`, but this specifically is checking if the request factory _can_ authenticate.

Comment (ryanisnan) [reply spylib/request.py]: What do you think?

**Comment by alekhrycaiko** [comment]
> @alekhrycaiko do you think we should bump the version? Probably a good habit to form.... slightly more annoying though :P

Ye, let's do a minor version bump, since this technically won't break anything.

File: spylib/request.py (hunk 3 above)
Comment (alekhrycaiko) [reply spylib/request.py]: I think this works. 🎏

**Pull request merged**